src/Patches.php

```diff
@@ -351,18 +351,21 @@ protected function getAndApplyPatch(RemoteFilesystem $downloader, $install_path,
       $downloader->copy($hostname, $patch_url, $filename, FALSE);
     }
 
-    // Modified from drush6:make.project.inc
-    $patched = FALSE;
-    // The order here is intentional. p1 is most likely to apply with git apply.
-    // p0 is next likely. p2 is extremely unlikely, but for some special cases,
-    // it might be useful.
-    $patch_levels = array('-p1', '-p0', '-p2');
-    foreach ($patch_levels as $patch_level) {
-      $checked = $this->executeCommand('cd %s && git --git-dir=. apply --check %s %s', $install_path, $patch_level, $filename);
-      if ($checked) {
-        // Apply the first successful style.
-        $patched = $this->executeCommand('cd %s && git --git-dir=. apply %s %s', $install_path, $patch_level, $filename);
-        break;
+    // Check if code is patched already.
+    $patched = $this->executeCommand('cd %s && git --git-dir=. apply -R --check -p1 %s', $install_path, $filename);
+
+    if (!$patched) {
+      // The order here is intentional. p1 is most likely to apply with git apply.
+      // p0 is next likely. p2 is extremely unlikely, but for some special cases,
+      // it might be useful.
+      $patch_levels = array('-p1', '-p0', '-p2');
```

Comment:
Just noticed why I did not try the other levels. I was too lazy to refractor exiting the loop in order to keep an other level from overriding a valid check. If nobody else needs this (its just an output cleaner) I'll come back to refractor later...